cloudformation/master-server-substack.cfn.yaml

```diff
@@ -480,55 +480,62 @@ Resources:
         deployConfigFiles:
           files:
             /tmp/dna.json:
-              mode: '000644'
-              owner: root
-              group: root
-              content:
-                cfncluster:
-                  stack_name: !Ref 'MainStackName'
-                  enable_efa: !Ref 'EFA'
-                  cfn_raid_vol_ids: !Ref 'RAIDVolIds'
-                  cfn_raid_parameters: !Ref 'RAIDOptions'
-                  cfn_scheduler_slots: !If
+              content: !Sub
```

Comment:
Just curious, why is `Sub` needed here? I guess we can't deploy file with `""` using just userdata?

Reply:
yes, the reason is "${SchedulerSlots}" and "${DisableHyperthreadingManually}" are two conditions. We can't deploy conditions directly and need !sub to define the value below. like
```
  !Sub
  - String
  - Var1Name: Var1Value
    Var2Name: Var2Value
```